signac/core/synced_collection.py

```diff
@@ -8,19 +8,132 @@
 backend with different data-structures or vice-versa. It declares as abstract
 methods the methods that must be implemented by any subclass to match the API.
 """
+import logging
 
-from typing import List, Callable
+from typing import List, Callable, DefaultDict, Any
 from contextlib import contextmanager
 from abc import abstractmethod
 from collections import defaultdict
 from collections.abc import Collection
 
+from .errors import Error
+
 try:
     import numpy
     NUMPY = True
 except ImportError:
     NUMPY = False
 
+logger = logging.getLogger(__name__)
+
+_BUFFERED_MODE = 0
+_BUFFERED_MODE_FORCE_WRITE = None
+_BUFFERED_BACKNDS: List[Any] = list()
+
+
+class BufferException(Error):
+    """An exception occured in buffered mode."""
+
+
+class BufferedError(BufferException):
+    """Raised when an error occured while flushing one or more buffered files.
+
+    Attribute
+    ---------
+    names:
```

Comment:
Doesn't match the implementation.
```suggestion
    files:
```